rust/noosphere-core/src/view/sphere.rs

```diff
@@ -606,8 +786,68 @@ impl<S: BlockStore> Sphere<S> {
                     _ => ()
                 };
 
-                yield (cid, Sphere::at(&cid, &self.store));
+                yield (cid, Sphere::from_memo(&memo, &self.store)?);
+            }
+        }
+    }
+
+    // Validate this sphere revision's signature and proof chain
+    pub async fn verify_signature(&self) -> Result<()> {
+        let memo = self.to_memo().await?;
+
+        // Ensure that we have the correct content type
+        memo.expect_header(&Header::ContentType, &ContentType::Sphere)?;
+
+        // Extract signature from the eponimous header
+        let signature_header = memo
+            .get_header(&Header::Signature)
+            .first()
+            .cloned()
+            .ok_or_else(|| anyhow!("No signature header found"))?;
+
+        let signature = base64_decode(&signature_header)?;
+
+        let mut did_parser = DidParser::new(SUPPORTED_KEYS);
+
+        let sphere_ipld = self.to_body().await?;
+
+        // If we have an authorizing proof...
+        if let Some(proof_header) = memo.get_header(&Header::Proof).first() {
+            // Interpret the header as a JWT-encoded UCAN..
+            let ucan = Ucan::try_from(proof_header.as_str())?;
+
+            // Discover the intended audience of the UCAN
+            let credential = did_parser.parse(ucan.audience())?;
+
+            // Verify the audience signature of the body CID
+            credential.verify(&memo.body.to_bytes(), &signature).await?;
+
+            // Check the proof's provenance and that it enables the signer to sign
+            let ucan_store = UcanStore(self.store.clone());
+            let proof = ProofChain::from_ucan(ucan, None, &mut did_parser, &ucan_store).await?;
```

Comment:
Do we need to consider publication time validity, like link records?

Reply:
Good catch, yes.

Reply:
(BTW, this is not new implementation; this was moved into `Sphere` but used to be a stand-alone function used in tests)